eng/common/TestResources/New-TestResources.ps1

```diff
@@ -133,6 +135,26 @@ if (!$templateFiles) {
     exit
 }
 
+$UserName =  if ($env:USER) { $env:USER } else { "${env:USERNAME}" }
+
+# If no base name is specified use current user name
+if (!$BaseName) {
+    $BaseName = "$UserName$ServiceDirectory"
+
+    Log "BaseName was not set. Using default base name: '$BaseName'"
+}
+
+# Try detecting repos that support OutFile and defaulting to it
+if (!$CI -and !$PSBoundParameters.ContainsKey('OutFile') -and $IsWindows)
+{
+    # TODO: find a better way to detect the language
```

Comment:
@chidozieononiwu is working on some common scripts one of which would give you the language. If you import [common.ps1](https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/common.ps1) it will import the [language-settings.ps1](https://github.com/Azure/azure-sdk-for-net/blob/master/eng/scripts/Language-Settings.ps1) which will have `$language` defined.

Reply:
Do you mind if I file this as a followup?

Reply:
Is that really the heuristic we want anyway? Ideally, eventually every language in any platform can do this. It's just a matter of how. When we three talked about this months ago, one idea was even just to chmod the file, which is all the CLI does on linux anyway. I'd rather we base this on some marker - perhaps a file under /eng, or something more general purpose we can read and getting repo configs, like a .json file that PS can read as an object. Then we don't have to keep updating this script and going through all the process around pushing it down.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/1082